### PR TITLE
Build stocks App using API

### PR DESCRIPTION
**Pull request by zunairkhan811**
- The project has been completed according to the requirement
- Navbar component is made
- StockList component is made
- StockDetails component is made
- Linters errors are removed
- When clicking on any stock it displays the data of that stock only
- A search bar is made which filters out the stocks
- project video is made and its link is placed in the readme.
- the project is deployed using GitHub pages

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,220 @@
...
+- [x] Show details of specific stock on click using API
+- [x] User can filter specific stock using searchbar
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live](https://zunairkhan811.github.io/react-stocks-project/)
```
Comment (FrederickMih) [inline README.md:75]: - [x] Let's kindly make sure our project is deployed properly. The home and detail page should work as expected. As you can see in the screenshot below, data was not fetched from the API.
Screenshot from 2023-05-25 19-41-19

File: README.md [hunk 2]
```diff
@@ -1 +1,220 @@
...
+
+## ⭐️ Show your support <a name="support"></a>
+
+If you like this feel free to follow me on github and give this project a star..
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
```
Comment (FrederickMih) [inline README.md:193]: - [x] As stated in the requirements, let's kindly give appropriate credit to the author of the original design.

File: src/components/StocksList.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
+import { useSelector, useDispatch } from 'react-redux';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faArrowAltCircleRight } from '@fortawesome/free-solid-svg-icons';
+import fetchStocksData from '../redux/stocks/fetchApi';
+
+const StocksList = () => {
```
Comment (FrederickMih) [inline src/components/StocksList.jsx:8]: - [x] App is not working properly as a result of data not being fetched properly from the API. Let's kindly make sure our app is working properly locally. Below is the screenshot of the error. You may once more verify if the request limit for the API has been reached. Your app needs to be working properly.
Screenshot from 2023-05-25 19-41-19

File: src/App.test.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
+  render(<App />);
+  const linkElement = screen.getByText(/learn react/i);
+  expect(linkElement).toBeInTheDocument();
+});
```
Comment (FrederickMih) [inline src/App.test.js]: - [x] Right now some of your tests are failing. You'll be equally required to fix them.
Screenshot from 2023-05-25 20-07-07

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @zunairkhan811, 


Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!



## Highlights :green_circle: 

- PR is professional :+1: 
- Good commit history :+1: 
- Linter checks passed :heavy_check_mark: 


## Required Changes ♻️

- [x] **Video Presentation:** Great job presenting the features of your project :+1: Now let's kindly record another video and highlight the piece of code we are proud of making as requested in the requirements.
- [ ] Kindly add tests for some functions in the Redux code and Integration tests using React Testing Library.



_Check the comments under the review._

## Optional suggestions
**N/A**


You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 1 above)
Comment (zunairkhan811) [reply README.md:75]: Working fine now

File: README.md (hunk 2 above)
Comment (zunairkhan811) [reply README.md:193]: Credit has been given

File: src/App.test.js (hunk 4 above)
Comment (zunairkhan811) [reply src/App.test.js]: Added more tests and test are working fine now

File: src/components/StocksList.jsx (hunk 3 above)
Comment (zunairkhan811) [reply src/components/StocksList.jsx:8]: Working fine now

**Review by Lizdev-05 — CHANGES_REQUESTED** [review]
Hi @zunairkhan811, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but some are still not addressed yet.



 ## Highlights
*  Brilliant implementation ✔️

* Tests are passing

*  PR is professional 👍

* Good commit history 👍

* Linter checks passed ✔️

 ## Required Changes ♻️

- [x] [REQUIRED]As requested by the previous reviewer, kindly show us a piece of code you are proud of, you showed us the design but not the code. The requirement says to present a piece of the code you finds interesting or proud of making. Kindly show us this in your video. 🥺 🙏 

## Optional suggestions
- N/A
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by addod19 — APPROVED** [review]
### Hi @zunairkhan811, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [x] Nice job showing a piece of code you are proud of making :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**